xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -769,8 +770,13 @@ void CGUIDialogPVRTimerSettings::InitializeTypesList()
     if (!type->IsTimerRule() && m_timerInfoTag->EndAsLocalTime() < CDateTime::GetCurrentDateTime())
       continue;
 
+    if (type == m_timerType)
```

Comment:
Please change to <code>if (!foundThisType && *type == *m_timertype)</code>.

First is for efficiency, to only compare types if actually needed.
Second is to not compare the timer type pointers but the actual objects.